openapi/index.yaml

```diff
@@ -15,35 +15,66 @@ schemes:
   - https
 paths:
   "/echo/{input}":
-    parameters:
-      - name: input
-        in: path
-        type: string
-        required: true
-        description: test input.
     get:
       operationId: echo
       summary: Echo API
-      description: Some API description here.
+      description: An echo API to check input
       responses:
         "200":
           description: Found.
-          schema:
-            "$ref": "#/definitions/Input"
-          examples:
-            application/json:
-              id: "10000AAAA"
-              output: "hello world !"
         "401":
           description: Unauthorized
         "404":
           description: Not found.
       parameters: []
 definitions:
-  Input:
+  Timestamp:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/v10.2.3/openapi/definitions.yaml#/Timestamp"
+  FiscalCode:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/v10.2.3/openapi/definitions.yaml#/FiscalCode"
+  ProblemJson:
+    $ref: "https://raw.githubusercontent.com/pagopa/io-functions-commons/v10.2.3/openapi/definitions.yaml#/ProblemJson"
+
+  CgnPendingStatus:
     type: object
     properties:
-      id:
+      status:
         type: string
-      output:
+        x-extensible-enum:
+          - PENDING
+    required:
+      - status
+  CgnActivatedStatus:
+    type: object
+    properties:
+      status:
+        type: string
+        x-extensible-enum:
+          - ACTIVATED
+    required:
+      - status
+  CgnRevokedStatus:
+    type: object
+    properties:
+      status:
+        type: string
+        x-extensible-enum:
+          - REVOKED
+    required:
+      - status
+  CgnCanceledStatus:
+    type: object
+    properties:
+      status:
         type: string
+        x-extensible-enum:
+          - CANCELED
+    required:
+      - status
+  CgnStatus:
+    x-one-of: true
+    allOf:
+      - $ref: "#/definitions/CgnPendingStatus"
+      - $ref: "#/definitions/CgnActivatedStatus"
+      - $ref: "#/definitions/CgnRevokedStatus"
+      - $ref: "#/definitions/CgnCanceledStatus"
```

Comment:
I understand the purpose, but I wonder which code will be generated from it. Can we try before commiting?

Reply:
Sure, the following is a sample:
```
import { Timestamp } from "./Timestamp";
import * as t from "io-ts";
import { enumType } from "italia-ts-commons/lib/types";

export enum StatusEnum {
  "ACTIVATED" = "ACTIVATED"
}

// required attributes
const CgnActivatedStatusR = t.interface({
  status: enumType<StatusEnum>(StatusEnum, "status")
});

// optional attributes
const CgnActivatedStatusO = t.partial({});

export const CgnActivatedStatus = t.intersection(
  [CgnActivatedStatusR, CgnActivatedStatusO],
  "CgnActivatedStatus"
);

export type CgnActivatedStatus = t.TypeOf<typeof CgnActivatedStatus>;
```

Reply:
mmm, what about `CgnStatus`? Won't be generated?

Reply:
Here we are :) 
```
import * as t from "io-ts";
import { CgnPendingStatus } from "./CgnPendingStatus";
import { CgnActivatedStatus } from "./CgnActivatedStatus";
import { CgnRevokedStatus } from "./CgnRevokedStatus";
import { CgnCanceledStatus } from "./CgnCanceledStatus";

export const CgnStatus = t.union(
  [CgnPendingStatus, CgnActivatedStatus, CgnRevokedStatus, CgnCanceledStatus],
  "CgnStatus"
);

export type CgnStatus = t.TypeOf<typeof CgnStatus>;
```

Reply:
Ok, so it's not an enum, but a union. Then I'm a bit concerned, as we cannot use `CgnStatus` as an enum. See [here](https://www.typescriptlang.org/play?#code/KYOwrgtgBAwg5iACqAJgSxHAygFwIY5gDOUA3lAA6oZxQC+AUKJLAgIIDGOaAbgcClwFiZKHi69+Keg1k4AnlVYghhEgF5lyEOkyriAH3ghO3PjgH6ishhwD2IIjihPhRAIxRNx7buz41ADoqHRooAHpwqBwACzQSeKg+ABs0FFsHJxcA4gAmL2UrYOpMCKjYxMSQOxxZIA).

In the end it's ok, I don't like enums. I just find it a bit messy, but that it's probably due to a lack con OpenAPI specification itself.